src/sbf.cpp

```diff
@@ -282,21 +294,25 @@ bool GPSDriverSBF::sendMessageAndWaitForAck(const char *msg, const int timeout)
 	return found_response;
 }
 
-// -1 = error, 0 = no message handled, 1 = message handled, 2 = sat info message handled
+// return value:
+// 0b1111_1111 = an error occurred
+// 0b0000_0000 = no message handled (not set up yet)
+// 0b0000_0001 = message handled
+// 0b0000_0010 = sat info message handled
```

Comment:
If that's the return type I would make it an enum with these values, e.g.

```suggestion
enum class ReceiveResult : uint8_-t {
    Error = 0b11111111,
    NoMessageHandled = 0,
    MessageHandled = 0b01,
    SatInfoMessageHandled = 0b10,
};
```

That's if we can actually change the interface of the receive function.

And `0b11111111` doesn't work well for `int` type, as it's not -1 in that case.

Reply:
I spent some time looking at this but changing the return type of `receive()` alone doesn't seem like a good idea as it uses some other functions that themselves return `int` bitfields. It would add a lot of casts back and forth between int and custom return types. I'm particularly worried about changing `payloadRxDone()` as its implementation isn't documented that well so I don't know what it is supposed to return in which case.

`receive()` is also part of the `GPSHelper` class which is implemented by quite a few other classes. I think it might be better to leave this as is. I can remove the change to the documentation above the implementation of `receive()` as it's not completely correct (e.g. the `0b1111_1111`).

**All the signatures that would need updates, including to the parent class in the case of `receive()`:**
![image](https://github.com/PX4/PX4-GPSDrivers/assets/160142177/9f3d89f9-f9a8-4fe0-9ff3-1b93949392b8)

PS: I think updating this code/cleaning it up a bit is a good idea after a decision is made about <https://github.com/PX4/PX4-Autopilot/pull/22904> as lots of code can be removed or moved to QGroundControl if that one is merged.

Reply:
Oh I see, this is a general interface! Let's leave it as is then. Sorry, I thought that was new.

Reply:
No problem. I'd be happy to have another look at it later and change the signature in `GPSHelper` if necessary, but for now I don't have the time to do these bigger refactoring tasks.